src/components/apps/listing/Listing.js

```diff
@@ -23,22 +41,192 @@ function Listing(props) {
+            setAllAppsKey([
+                "getApps",
+                {
+                    rowsPerPage,
+                    orderBy,
+                    order,
+                    page,
+                    appTypeFilter,
+                },
+            ]);
+            setAppsInCategoryKey(null);
+        } else if (systemId && categoryId) {
             setAppsInCategoryKey([
                 "getAppsInCategory",
-                { systemId, rowsPerPage, orderBy, order, page, categoryId },
+                {
+                    systemId,
+                    rowsPerPage,
+                    orderBy,
+                    order,
+                    page,
+                    categoryId,
+                    appTypeFilter,
+                },
             ]);
+            setAllAppsKey(null);
+        }
+    }, [order, orderBy, page, rowsPerPage, selectedCategory, filter]);
+
+    useEffect(() => {
+        if (data && data.Location && data.status === 302) {
+            setAgaveAuthDialogOpen(true);
+        }
+    }, [data, setAgaveAuthDialogOpen]);
+
+    useEffect(() => {
+        if (detailsOpen && data && data.apps) {
+            const selectedId = selected[0];
+            const appIndex = data.apps.findIndex(
+                (item) => item.id === selectedId
+            );
+            setDetailsApp(data.apps[appIndex]);
         }
-    }, [order, orderBy, page, rowsPerPage, selectedCategory]);
+    }, [data, detailsOpen, selected]);
+
+    useEffect(() => {
+        setDetailsEnabled(selected && selected.length === 1);
+    }, [selected]);
+
+    useEffect(() => {
+        if (detailsApp) {
+            setDetailsKey([
+                "getAppsDetails",
+                {
+                    systemId: detailsApp.system_id,
+                    appId: detailsApp.id,
+                },
+            ]);
+        }
+    }, [detailsApp]);
+
+    useEffect(() => {
+        if (appsInCategoryError || listingError) {
+            announce({
+                text: formatMessage(intl, "appListingError"),
+                variant: AnnouncerConstants.ERROR,
+            });
+            setData(null);
+        }
+        if (detailsError) {
+            announce({
+                text: formatMessage(intl, "appDetailsError"),
+                variant: AnnouncerConstants.ERROR,
+            });
+            setDetails(null);
+        }
+        if (favMutationError) {
+            announce({
+                text: formatMessage(intl, "favMutationError"),
+                variant: AnnouncerConstants.ERROR,
+            });
+        }
+        if (ratingMutationError) {
+            announce({
+                text: formatMessage(intl, "ratingMutationError"),
+                variant: AnnouncerConstants.ERROR,
+            });
+        }
+    }, [
+        appsInCategoryError,
+        listingError,
+        detailsError,
+        favMutationError,
+        ratingMutationError,
+        intl,
+    ]);
```

Comment:
This code won't until we fix the `TODO` comment in `callAPI`. That is remove all loading and error handling logic.

Reply:
Yes - you can use storybook knobs for almost anything. I make heavy usage of them in the DataDetails story. 
You could probably do something like
```
import { boolean } from "@storybook/addon-knobs";

const success = boolean("Use success responses", true);
mockAxios
    .onGet(/whatever/)
    .reply(
        success ? 200 : 500,
        success ? successResp : errorResp
    );
```

Reply:
I couldn't get mockAxios to work with `success` / `failure` boolean knob. Regardless, I have refactored the code so each error has its own `useEffect`.